src/lib/tabs/tab-header.ts

```diff
@@ -0,0 +1,258 @@
+import {
+  ViewChild,
+  Component,
+  Input,
+  NgZone,
+  QueryList,
+  ElementRef,
+  Renderer, ViewEncapsulation, ContentChildren, Output, EventEmitter
+} from '@angular/core';
+import {RIGHT_ARROW, LEFT_ARROW, ENTER, Dir, LayoutDirection} from '../core';
+import {MdTabLabelWrapper} from './tab-label-wrapper';
+import {MdInkBar} from './ink-bar';
+import 'rxjs/add/operator/map';
+
+export type ScrollDirection = 'after' | 'before';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-tab-header',
+  templateUrl: 'tab-header.html',
+  styleUrls: ['tab-header.css'],
+  encapsulation: ViewEncapsulation.None,
+  host: {
+    '[class.md-tab-header]': 'true',
+    '[class.scroll]': '_isScrollingEnabled()',
```

Comment:
Yeah, I chatted with Andrew yesterday about this and that's pretty much example what we're doing. MutationObserver per-tab, going to do that part in a follow-up PR (and just handle init, adding/removing tabs in this PR).